packages/core/modules/sun_tracking.py

```diff
@@ -76,32 +74,55 @@ def run(self, new_config: types.ConfigDict) -> None:
 
         if (time.time() - self.last_start_time) < 300:
             logger.debug(
-                "Skipping motor validation when CamTracker "
-                + "has been started less than 5 minutes ago"
+                "Skipping CamTracker state validation when " +
+                "it has been started less than 5 minutes ago"
             )
             return
 
-        tracker_position_is_valid = self.validate_tracker_position()
-        if tracker_position_is_valid is None:
-            logger.debug("CamTracker motor position is unknown (last log line too old).")
-        else:
-            if tracker_position_is_valid:
-                logger.debug("CamTracker motor position is valid.")
-            else:
-                logger.info("CamTracker motor position is over threshold.")
-                logger.info("Stopping CamTracker. Preparing for reinitialization.")
-                self.stop_sun_tracking_automation()
+        # check the current positions of the CamTracker motors
+        # and the TUM PLC cover and restart CamTracker if they
+        # are not in the expected position
+        restart_camtracker: bool = False
+
+        logger.debug("Checking CamTracker motor position")
+        motor_position_state = self.check_camtracker_motor_position()
+
+        if self.config.tum_plc is not None:
+            logger.debug("Checking TUM enclosure cover position")
+            cover_position_state = self.check_tum_plc_cover_position()
+            if cover_position_state == "angle not reported":
+                logger.debug("TUM enclosure cover position is unknown.")
+            if cover_position_state == "invalid":
+                logger.info("TUM enclosure cover is closed.")
+                if self.config.camtracker.restart_if_cover_remains_closed:
+                    restart_camtracker = True
+            if cover_position_state == "valid":
+                logger.debug("TUM enclosure cover is open.")
+
+        if motor_position_state == "logs too old":
+            logger.debug("CamTracker motor position is unknown (no log line).")
+            if self.config.camtracker.restart_if_logs_are_too_old:
+                restart_camtracker = True
+        if motor_position_state == "invalid":
+            logger.info("CamTracker motor position is over threshold.")
+            restart_camtracker = True
+        if motor_position_state == "valid":
+            logger.debug("CamTracker motor position is valid.")
+
+        if restart_camtracker:
```

Comment:
We might need to add a timer here. CamTracker Shutdown and Init take some time and if the loop revisits this part too fast it might end up in an infinite restart process.

Reply:
Same as in the [comment below](https://github.com/tum-esm/pyra/pull/197#discussion_r1389727817).

This validation only start after CamTracker has been running for at least 5 minutes.